docs/examples/handling_custom_types/test_example_2.py

```diff
@@ -0,0 +1,53 @@
+from dataclasses import dataclass
+from typing import Type, Dict, Any, TypeVar, Generic
+from uuid import UUID
+
+from polyfactory.factories import DataclassFactory
+
+
+# we created a special class we will use in our code
+class CustomSecret:
+    def __init__(self, value: str) -> None:
+        self.value = value
+
+    def __repr__(self) -> str:
+        return "*" * len(self.value)
+
+    def __str__(self) -> str:
+        return "*" * len(self.value)
+
+
+T = TypeVar("T")
+
+
+# we create a custom base factory to handle dataclasses, with an extended provider map
+class CustomDataclassFactory(Generic[T], DataclassFactory[T]):
+    __is_base_factory__ = True
+
+    @classmethod
+    def get_provider_map(cls) -> Dict[Type, Any]:
+        providers_map = super().get_provider_map()
+
+        return {
+            CustomSecret: lambda: CustomSecret("jeronimo"),
+            **providers_map,
+        }
+
+
+@dataclass
+class Person:
+    id: UUID
+    secret: CustomSecret
+
+
+# we use our CustomDataclassFactory as a base for the PersonFactory
```

Comment:
sorry i was late on this, looks good - curious on how the test example script comments look or if it is just for internal use